src/pinns_pde_solve.jl

```diff
@@ -185,6 +185,9 @@ where
 function _transform_expression(ex,dict_indvars,dict_depvars,chain,eltypeθ,strategy)
     _args = ex.args
     for (i,e) in enumerate(_args)
+        if e in [+,-,*,/,^,inv]
```

Comment:
why is this necessary?

Reply:
Inorder to disable `__dot__` for specific functions.

Reply:
but why these functions, and why here? I don't get it.

Reply:
sorry for the long answer, 
the problem is that `toexpr` transforms not as symbols but as functions of all arithmetic operators. But in Broadcast.jl the function is used as a marker for not dotting (which is generally weird)
https://github.com/JuliaLang/julia/blob/4270d3b44e53e9fa68dcdbb6cb416d232615db4c/base/broadcast.jl#L1295


```julia
@parameters x y
@variables u(..)
eq =  u(x)+ u(x)*x
expr = toexpr(eq)
julia> dump(expr)
Expr
  head: Symbol call
  args: Array{Any}((3,))
    1: + (function of type typeof(+))
    ...

Base.Broadcast.dottable(x::Function) = false
julia> Broadcast.__dot__(expr)
:((+)(u.(x), (*)(x, u.(x))))

Base.Broadcast.dottable(x::Function) = true
julia> Broadcast.__dot__(expr)
:((+).(u.(x), (*).(x, u.(x))))
```

Reply:
so to avoid piracy  and broadcasted full expression with arithmetic in need convert them in symbol

Reply:
in general, this is a crutch, using functions inside an expression to avoid dotting. 
https://github.com/JuliaLang/julia/blob/4270d3b44e53e9fa68dcdbb6cb416d232615db4c/base/broadcast.jl#L1295

Reply:
leading to the fact that we have to exclude functions in order to correct broadcast expr, which is kind of bad

Reply:
```suggestion
        if e isa Function
```

?

Reply:
it doesn't work,   `Differential` is also function( .
```julia
Dθ = Differential(θ)
eq = Dθ(u(θ))
expr = toexpr(eq)
julia> dump(expr)
Expr
  head: Symbol call
  args: Array{Any}((2,))
    1: Differential(θ) (function of type Differential)
...
```

Reply:
to avoid all this, in the end, we need our own `toexpr` in NeuralPDE

Reply:
`if e isa Function && e !== Differential`

Reply:
>to avoid all this, in the end, we need our own toexpr in NeuralPDE

That seems unnecessary.

Reply:
Why would it require an entire new `toexpr` instead of just omitting one function?

Reply:
Since `ModelingToolKit.toexpr` does not guarantee the same behavior, many times it was necessary to rewrite the parser after updating ModelingToolKit.jl

but it’s not urgent, as long as it works)